### ISSUE TITLE
How to save FCM token with javascriptinterface

### ISSUE THREAD

**Issue by paundri** [mgks/Android-SmartWebView#213]
My Webview is already connect to FCM, in logcat I already have the FCM device id token, and also I manage to enable javascriptinterface, but I cannot take the fcm token with javascriptinterface....how to manage that ?

**Comment by mgks** [comment]
if you can read the cookies with javascript.

**Comment by paundri** [comment]
Yes, First I save the token on sharedpreference and then I took the value with javascriptinterface,

**Comment by paundri** [comment]
Now the problem is fcm notification only work on background and foreground, but when app is killed the notification never arrive?....any idea how to solve it ? ...I allready modify onMessageReceived method...but the problemm is still persist

**Comment by natsirasrafi** [comment]
private final class WebInterface {

		private final WeakReference<Activity> mContextRef;
		public WebInterface(Activity context) {
			this.mContextRef = new WeakReference<Activity>(context);
		}
	
		@JavascriptInterface
		public String gettokenfcm() {
			FirebaseInstanceId.getInstance().getInstanceId().addOnSuccessListener( MainActivity.this, instanceIdResult -> {
				fcm_token = instanceIdResult.getToken();
			}).addOnFailureListener(e -> Log.d("REQ_FCM_TOKEN", "FAILED"));
			return fcm_token;
		}

	
	}

**Comment by mgks** [comment]
Fixed with #318
